apps/common/src/python/mediawords/util/mail.py

```diff
@@ -117,6 +117,9 @@ def send_email(message: Message) -> bool:
             message_part = MIMEText(message.text_body, 'plain', 'utf-8')
             mime_message.attach(message_part)
 
+        mime_message.add_header('List-Unsubscribe',
```

Comment:
Could you add a quick configuration parameter to make the email address in `List-Unsubscribe` configurable? We used to be the sole user of Media Cloud's backend so back in the day we would just hardcode our emails here and there (and they're still hardcoded at some places), but now Code for Africa people are running an instance, and I think there are at least two more backend instances running out there in the wild. So, we don't want to be getting their unsubscribe emails.

We currently store all (I think) of our configuration in environment variables (for better or for worse). We also try our best to keep that fact isolated from the rest of the system, i.e. instead of reading the required configuration parameter directly via, say, `os.environ`, we encapsulate the configuration either in "common" configuration (available in `mediawords.util.config.common`), or in a per-app configuration (here's a [sample Facebook client configuration](https://github.com/mediacloud/backend/blob/master/apps/facebook-fetch-story-stats/src/python/facebook_fetch_story_stats/config.py)), the reasons being:

* When you have a class between the configuration user and the environment variable, you get a chance to process the env. variable a bit, e.g. cast a number to `int`, cast `0` / `1` to `bool`, parse some sort of an URI, do validation, complain when the env. variable is not set, etc.
* Storing configuration (including secrets) in environment variables is not terrible but not great either, so if we ever come up with a better way to do this (Docker secrets? etcd or something?), we won't have to fish out all usages of environment variables in the whole Python codebase.

So that's why you can't just do `os.environ` :)

To add a new "common" (i.e. potentially used by all "apps") configuration parameter, you'll have to:

* Add a new method to [`SMTPConfig`](https://github.com/mediacloud/backend/blob/17bae7d31b294f92710387978e3cb6ed89df54bd/apps/common/src/python/mediawords/util/config/common.py#L126) (or some other class if that makes more sense to you) that would read the environment variable with your "send unsubscribe emails to this address" value
    * Consider making use of [`env_value()`](https://github.com/mediacloud/backend/blob/17bae7d31b294f92710387978e3cb6ed89df54bd/apps/common/src/python/mediawords/util/config/__init__.py#L18) to save up on some boilerplate for testing whether the env. variable exists, etc.
    * Make it an optional configuration parameter (i.e. if it doesn't get set, things don't crash) but make the default email address to a nonexistent email inbox (assuming that our partners will forget to set this property, we don't want their emails defaulting to our `support@`)
    * Optionally check whether what gets set looks like an email address by verifying that it has `@` in it (there are various ways for testing email address validity but there's no point bothering ourselves with that here as this is an internal configuration variable)
* Whenever you want to use it, just import `SMTPConfig` and call the static method that you've just created to get the email address

The email subject can stay non-configurable.

Reply:
Oh yeah, last step: you'll have to add the newly added configuration variable to both `apps/docker-compose.dist.yml` and `apps/common/docker-compose.tests.yml`. The `docker-compose.dist.yml` entry should have a comment for what this env. variable is there for and have a sample value (possibly commented out), and the `docker-compose.tests.yml` might have a similar commented out entry just to let the user (the future us basically) know that such an configuration environment variable exists and that it's supposed to do something.

I might be sounding extra nitpicky, but I think it's a good practice to keep a record of how each application is supposed to be configured.